python/ray/dag/compiled_dag_node.py

```diff
@@ -1660,18 +1798,21 @@ def _generate_dag_operation_graph_node(
         """
         from ray.dag.collective_node import CollectiveOutputNode, _CollectiveOperation
 
+        OpType = _DAGNodeOperationType
+
         assert self.idx_to_task
         assert self.actor_to_executable_tasks
 
+        for _, exec_tasks in self.actor_to_executable_tasks.items():
+            for exec_task in exec_tasks:
+                exec_task.synchronous_peer_idxs = [
```

Comment:
Can we put this in exec_task's constructor?

Reply:
Done.